src/providers/twitch/TwitchChannel.cpp

```diff
@@ -801,13 +804,39 @@ void TwitchChannel::loadRecentMessages()
                 }
             }
 
-            postToThread(
-                [shared, messages = std::move(allBuiltMessages)]() mutable {
-                    shared->addMessagesAtStart(messages);
-                });
+            postToThread([this, shared, root,
+                          messages = std::move(allBuiltMessages)]() mutable {
+                shared->addMessagesAtStart(messages);
+
+                // Notify user about a possible gap in logs
+                if (QString errorCode = root.value("error_code").toString();
+                    !errorCode.isEmpty())
+                {
+                    qCDebug(chatterinoTwitch)
+                        << QString("rm error_code=%1, channel=%2")
+                               .arg(errorCode, this->getName());
+                    if (errorCode == "channel_not_joined")
+                    {
+                        shared->addMessage(makeSystemMessage(
+                            "Recent-messages API isn't currently joined to "
+                            "this channel (in progress or due to an issue)."));
+                    }
+                }
+            });
 
             return Success;
         })
+        .onError([weak](NetworkResult result) {
+            auto shared = weak.lock();
+            if (!shared)
+                return;
+
+            shared->addMessage(makeSystemMessage(
+                QString("Recent-messages API responded with an error %1 while "
+                        "retrieving messages. This isn't Chatterino's fault, "
+                        "but rather a temporary issue with the service itself.")
```

Comment:
I don't think we should play the blame game here and drop the phrase "This isn't Chatterino's fault"

Reply:
I think the message `Message history service unavailable (Code: %1)` would be descriptive enough, and give service owners & us enough information to debug